src/main/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketPluginConfiguration.java

```diff
@@ -29,13 +33,13 @@ public BitbucketPluginConfiguration() {
     }
 
     @Override
-    public boolean configure(StaplerRequest req, JSONObject json) throws FormException {
+    public boolean configure(StaplerRequest req, JSONObject json) {
         req.bindJSON(this, json);
-        if (getServerList().stream().allMatch(server -> server.validate())) {
-            save();
-            return true;
-        }
-        return false;
+        FormValidation aggregate = FormValidation.aggregate(serverList.stream()
+                .map(BitbucketServerConfiguration::validate)
+                .collect(Collectors.toList()));
+        save();
```

Comment:
Should we only be saving if the validation passes?

Reply:
I'm open to suggestions. I opted to save regardless because we construct a new `BitbucketPluginConfiguration` whenever we configure the SCM, which calls `load` on initialization, while the config screen always uses the same instance, so there is the potential for the two to fall out of sync. Given we hide invalid server configurations anyway and you would have to restart jenkins or edit the `config.xml` directly to preserve your old settings, I think the consistency in representations is more valuable here.
That said, it might be better still if the SCM and settings screen are using the same plugin configuration object. Not sure if we have access to it but if this is considered an issue, I'm happy to look into it.

Reply:
Disregard everything I just said, will rebase on master and only use the one `BitbucketPluginConfiguration`. Sorry!